guides/micronaut-data-jdbc-repository/micronaut-data-jdbc-repository.adoc

```diff
@@ -1,47 +1,63 @@
 include::{commondir}/common-header.adoc[]
 
-The application exposes some REST endpoints and stores data in a database using Micronaut Data JDBC.
+The application exposes some REST endpoints and stores data in a MySQL database using Micronaut Data JDBC.
 
-include::{commondir}/common-requirements.adoc[]
+== What you will need
+
+To complete this guide, you will need the following:
+
+* Some time on your hands
+* A decent text editor or IDE
+* JDK 1.8 or greater installed with `JAVA_HOME` configured appropriately
+* https://www.docker.io/gettingstarted/#h_installation[Docker] installed to run MySQL and for running tests, using https://www.testcontainers.org[Testcontainers].
 
 include::{commondir}/common-completesolution.adoc[]
 
-include::{commondir}/common-create-app.adoc[]
+== Writing the Application
 
-=== Data Source configuration
+include::{commondir}/common-cli-or-launch.adoc[]
 
-Add the following dependencies:
+:exclude-for-languages:groovy
+[source,bash]
+----
+mn create-app --features=data-jdbc,flyway,jdbc-hikari,mysql,testcontainers,graalvm example.micronaut.micronautguide --build=@build@ --lang=@lang@
+----
+:exclude-for-languages:
 
-:dependencies:
+:exclude-for-languages:java,kotlin
+[source,bash]
+----
+mn create-app --features=data-jdbc,flyway,jdbc-hikari,mysql,testcontainers example.micronaut.micronautguide --build=@build@ --lang=@lang@
+----
+:exclude-for-languages:
 
-dependency:micronaut-data-processor[groupId=io.micronaut.data,scope=annotationProcessor,callout=1]
-dependency:micronaut-data-jdbc[groupId=io.micronaut.data,callout=2]
-dependency:micronaut-jdbc-hikari[groupId=io.micronaut.sql,callout=3]
-dependency:h2[groupId=com.h2database,scope=runtimeOnly,callout=4]
+include::{commondir}/common-build-lang-arguments.adoc[]
 
-:dependencies:
+:exclude-for-languages:groovy
+If you use Micronaut Launch, select Micronaut Application as application type and add the `data-jdbc`, `flyway`, `jdbc-hikari`, `mysql`, `testcontainers` and `graalvm` features.
```

Comment:
we should be able to use `@features@` instead of the hardcoded list of feautres.

Reply:
this should use the diffLink macro

Reply:
and the diffLink macro should remove `graalvm` if the language is Groovy